### PR TITLE
Microsoft Simulator Library Fix

### PR DESCRIPTION
**Pull request by ScottCarda-MS**
Ensures the Q# libraries get regenerated with monomorphized content when compiling with the AzureSimulator process architecture.

Addressed #1069

### CODE REVIEW

File: src/QsCompiler/CSharpGeneration/Context.fs [hunk 1]
```diff
@@ -152,6 +152,7 @@ type CodegenContext =
                 target = AssemblyConstants.HoneywellProcessor
                 || target = AssemblyConstants.IonQProcessor
                 || target = AssemblyConstants.QCIProcessor
+                || target = "AzureSimulator" // ToDo: We need to have an assembly constant for this.
```
Comment (bettinaheim) [inline src/QsCompiler/CSharpGeneration/Context.fs:155]:
```suggestion
                || target = "MicrosoftSimulator" // ToDo: We need to have an assembly constant for this.
```

**Comment by bettinaheim** [comment]
@ScottCarda-MS Please either merge this to cesarcz/merge-feature-azure-simulator-to-main, or directly to main after that one went in.

File: src/QsCompiler/CSharpGeneration/Context.fs (hunk 1 above)
Comment (ScottCarda-MS) [reply src/QsCompiler/CSharpGeneration/Context.fs:155]: I'm not sure this is right. While debugging the name that came up was "AzureSimulator", not "MicrosoftSimulator".

Comment (bettinaheim) [reply src/QsCompiler/CSharpGeneration/Context.fs:155]: This is correct.

**Pull request merged**